### PR TITLE
*: improve deployment doc and toc

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

- Improves some description and cross-references in [Deploy a TiDB Cluster Using TiUP | TiDB Docs](https://docs.pingcap.com/tidb/dev/production-deployment-using-tiup)
  - Adds link of [Scale a TiDB Cluster Using TiUP | TiDB Docs](https://docs.pingcap.com/tidb/dev/scale-tidb-using-tiup) to the bottom section 
  - Adds link of [Best Practices for TiDB Security Configuration | TiDB Docs](https://docs.pingcap.com/tidb/dev/best-practices-for-security-configuration) to the beginning section
- Updates wording in `TOC.md` to make it consistent with the actual `title` in `.md` files

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.5 (TiDB 8.5 versions)
- [ ] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: production-deployment-using-tiup.md [hunk 1]
```diff
@@ -6,16 +6,20 @@ aliases: ['/docs/dev/production-deployment-using-tiup/','/docs/dev/how-to/deploy
...
 
-[TiUP](https://github.com/pingcap/tiup) is a cluster operation and maintenance tool introduced in TiDB 4.0. TiUP provides [TiUP cluster](https://github.com/pingcap/tiup/tree/master/components/cluster), a cluster management component written in Golang. By using TiUP cluster, you can easily perform daily database operations, including deploying, starting, stopping, destroying, scaling, and upgrading a TiDB cluster, and manage TiDB cluster parameters.
+This guide describes how to deploy a TiDB Self-Managed cluster using [TiUP](https://github.com/pingcap/tiup) in the production environment.
 
-TiUP supports deploying TiDB, TiFlash, TiCDC, and the monitoring system. This document introduces how to deploy TiDB clusters of different topologies.
+TiUP is a cluster operation and maintenance tool introduced in TiDB 4.0. It provides [TiUP cluster](https://github.com/pingcap/tiup/tree/master/components/cluster), a Golang-based component for managing TiDB clusters. By using the TiUP cluster, you can easily perform routine database operations, such as deploying, starting, stopping, destroying, scaling, upgrading TiDB clusters, and managing TiDB cluster parameters.
```
Comment (Oreoxmt) [inline production-deployment-using-tiup.md:11]:
```suggestion
TiUP is a cluster operation and maintenance tool introduced in TiDB v4.0. It provides [TiUP cluster](https://github.com/pingcap/tiup/tree/master/components/cluster), a Golang-based component for managing TiDB clusters. By using the TiUP cluster, you can easily perform routine database operations, such as deploying, starting, stopping, destroying, scaling, upgrading TiDB clusters, and managing TiDB cluster parameters.
```

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: hardware-and-software-requirements.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
 ---
-title: Software and Hardware Recommendations
-summary: Learn the software and hardware recommendations for deploying and running TiDB.
+title: TiDB Software and Hardware Requirements
+summary: Learn the software and hardware requirements for deploying and running TiDB.
 aliases: ['/docs/dev/hardware-and-software-requirements/','/docs/dev/how-to/deploy/hardware-recommendations/']
 ---
 
-# Software and Hardware Recommendations
+# TiDB Software and Hardware Requirements
```
Comment (lilin90) [inline hardware-and-software-requirements.md:7]: Note: This is to keep wording consistent with that in https://docs.pingcap.com/tidb/dev/check-before-deployment and that in Chinese.

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**